modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/Filters.tsx

```diff
@@ -306,13 +308,43 @@ function Filters({
 	};
 
 	const onCreationButtonClick = (filterType: EFilterType) => {
-		const availableFieldsList = fields.filter((item) =>
-			FILTER_TYPES[filterType as EFilterType].availableFieldsFilter(item)
-		);
+		let availableFieldsListLength = 0;
+
+		if (Liferay.FeatureFlags['LPD-25905']) {
+			const availableFfilterTypeFields = JSON.parse(
```

Comment:
Need to create a copy of the fields to use with the different types of filters as we only get the fields once and we need to "mutate" in order to identify which fields to enable for each type.

`visit` helper uses `forEach`  and returns the updated copy of the fields. As another option, if we want to remove the fields that do not match the filter criteria we'll need to create a custom utility.

Reply:
The problem only occurs with the FF (future default) as we need to go through the tree and identify which fields should be enabled/disabled based on the filter type.

With the previous implementation we were using `Array.filter` and we don't have this problem.